### PR TITLE
Make missing metadata images always invalid

### PR DESCRIPTION
**Pull request by kenoir**
... regardless of leasing status.

screenshot from 2016-08-16 14-57-21

### CODE REVIEW

File: media-api/app/lib/ImageExtras.scala [hunk 1]
```diff
@@ -40,29 +40,46 @@ object ImageExtras {
...
+    !optToBool(validityMap.find(_._2.valid == true))
+
+  def invalidityIsOverridden(
+    validityMap: Map[String, ValidityCheck],
+    validityOverrides: Map[String, Boolean]
+  ): Boolean = {
+      optToBool(validityOverrides.find(_._2 == true)) &&
+      !optToBool(validityMap.filter(_._2.valid == true)
+        .find(_._2.overrideable == false))
+  }
```
Comment (jennysivapalan) [inline media-api/app/lib/ImageExtras.scala]: This is a bit hard to follow, is it worth a unit test to help with the logic?

File: media-api/app/lib/ImageExtras.scala [hunk 2]
```diff
@@ -40,29 +40,46 @@ object ImageExtras {
...
+
+  def validityMap(image: Image): Map[String, ValidityCheck] = Map(
+    "paid_image"           -> ValidityCheck(Costing.isPay(image.usageRights), true),
+    "conditional_paid"     -> ValidityCheck(Costing.isConditional(image.usageRights), true),
+    "no_rights"            -> ValidityCheck(!hasRights(image.usageRights), true),
+    "missing_credit"       -> ValidityCheck(!hasCredit(image.metadata), false),
+    "missing_description"  -> ValidityCheck(!hasDescription(image.metadata), false),
+    "current_deny_lease"   -> ValidityCheck(hasCurrentDenyLease(image.leases), true),
+    "over_quota"           -> ValidityCheck(UsageQuota.isOverQuota(image.usageRights), true)
   )
```
Comment (jennysivapalan) [inline media-api/app/lib/ImageExtras.scala]: Why do some things get an override?

File: media-api/app/lib/ImageExtras.scala (hunk 1 above)
Comment (kenoir) [inline media-api/app/lib/ImageExtras.scala]: My instinct is adding a test is adding _more_ complicated code to test something already complicated. A test might help catch changes in behaviour if someone comes along and breaks it by accident.

I'd prefer to make the code clearer (and probably more verbose) to help with both things.

File: media-api/app/lib/ImageExtras.scala (hunk 2 above)
Comment (kenoir) [inline media-api/app/lib/ImageExtras.scala]: The logic is:
- Images can be invalid -> Invalid images can't be cropped
- Images can be deemed invalid by a set of checks
- Image validity can be overridden (to valid) if any one of a set of checks is met
- Some of the set of checks that mark an image as invalid cannot be overridden

File: media-api/app/lib/ImageExtras.scala [hunk 3]
```diff
@@ -13,56 +13,59 @@ object ImageExtras {
...
+      "conditional_paid"     -> createCheck(Costing.isConditional(image.usageRights)),
+      "no_rights"            -> createCheck(!hasRights(image.usageRights)),
+      "missing_credit"       -> createCheck(!hasCredit(image.metadata), false),
+      "missing_description"  -> createCheck(!hasDescription(image.metadata), false),
+      "current_deny_lease"   -> createCheck(hasCurrentDenyLease(image.leases)),
+      "over_quota"           -> createCheck(UsageQuota.isOverQuota(image.usageRights))
+    )
+  }
 
   def validityOverrides(image: Image, withWritePermission: Boolean): Map[String, Boolean] = Map(
```
Comment (jennysivapalan) [inline media-api/app/lib/ImageExtras.scala]: I think `validityOverrides` should move up to be above where `shouldOverride` is defined then it makes it a bit easier to follow the logic of createCheck method. Or it should be inlined with `shouldOverride` (as that is a val so called just once).

**Comment by jennysivapalan** [comment]
Definitely much simpler to follow 👍
